stacklight_tests/tests/test_smoke.py

```diff
@@ -1,7 +1,105 @@
+import pytest
+
 from stacklight_tests.tests import base_test
 
 
 class TestSmoke(base_test.BaseLMATest):
+
+    def test_influxdb_installed(self):
+        """Smoke test that checks basic features of InfluxDb.
+
+        Scenario:
+            1. Check InfluxDB package is installed
+            2. Check InfluxDB is up and running
+            3. Check that InfluxDB is online and can serve requests
+
+        Duration 1m
+        """
+        service = "influxdb"
+        self.check_service_installed(service)
+        self.check_service_running(service)
+        measurements, env_name = self.influxdb_api.check_influxdb_online()
+        assert measurements and env_name
+
+    def test_grafana_installed(self):
+        """Smoke test that checks basic features of Grafana.
+
+        Scenario:
+            1. Check Grafana package is installed
+            2. Check Grafana is up and running
+            3. Check that user can login into and HTTP API is working
+            4. Check that access prohibited for non-authorized user
+
+        Duration 1m
+        """
+        self.check_service_installed("grafana")
+        self.check_service_running("grafana-server")
+        self.grafana_api.check_grafana_online()
+
+    def test_nagios_installed(self):
+        """Smoke test that checks basic features of Nagios.
+
+        Scenario:
+            1. Check that hosts page is available
+            2. Check that services page is available
+            3. Check that access prohibited for non-authorized user
+
+        Duration 1m
+        """
+        hosts = self.nagios_api.get_all_nodes_statuses()
+        services = self.nagios_api.get_all_services_statuses()
+        assert hosts and services
+        # Negative testing
+        origin_password = self.nagios_api.password
+
+        def set_origin_password():
+            self.nagios_api.password = origin_password
+            self.nagios_api.nagios_url = self.nagios_api.format_url()
+        self.destructive_actions.append(set_origin_password)
+        self.nagios_api.password = "rogue"
+        self.nagios_api.nagios_url = self.nagios_api.format_url()
```

Comment:
I think it's OK, because I need to provide callable to destructive actions, but I don't want to write there an anonymous function. Also it is used only for internal purposes of current test.